proposed/2021/NuGetConsumePDBInPackageReference.md

```diff
@@ -0,0 +1,216 @@
+NuGet will check and add all extension, no matter the extension is in [AllowedReferenceRelatedFileExtensions](https://github.com/dotnet/msbuild/blame/main/src/Tasks/Microsoft.Common.CurrentVersion.targets#L621-L627) or not. So if there is a `a.random`, NuGet will add `random` into the `related` property.
+
+### 3. Will having `related` property make any difference in No-op restore? 
+For a package reference, there should be no change in `related` property unless we change to reference a different package/version. So there is no difference in No-op restore when having `related` property.
+
+## Considerations
+
+### In addition to the proposed approach, 2 of other solutions were considered. 
+
+#### **Do nothing - Recommend the customers use the custom target workaround**
+
+The workaround that copying .pdb and .xml file into build output folder(.NET Framework) is as following:
+```xml
+ <Target Name="_ResolveCopyLocalNuGetPackagePdbsAndXml" 
+          Condition="$(CopyLocalLockFileAssemblies) == true" 
+          AfterTargets="ResolveReferences">
+    <ItemGroup>
+      <ReferenceCopyLocalPaths 
+        Include="@(ReferenceCopyLocalPaths->'%(RootDir)%(Directory)%(Filename).pdb')" 
+        Condition="'%(ReferenceCopyLocalPaths.NuGetPackageId)' != ''
+                    and Exists('%(RootDir)%(Directory)%(Filename).pdb')" />
+      <ReferenceCopyLocalPaths 
+        Include="@(ReferenceCopyLocalPaths->'%(RootDir)%(Directory)%(Filename).xml')" 
+        Condition="'%(ReferenceCopyLocalPaths.NuGetPackageId)' != ''
+                    and Exists('%(RootDir)%(Directory)%(Filename).xml')" />
+    </ItemGroup>
+  </Target>
+```
+Pros:
+* It works (in certain condition).
+
+Cons:
+* It only works for .NET Framework. For .NET Core, it needs to be adjusted.
+* It only copies .pdb and .xml file to build output. If publish output path is different from build output path, it needs to be adjusted to publish .pdb and .xml.
+
+
+#### **Add as LockFileItem, but not a property of LockFileItem**
+
+Instead of adding existing extension as a property of LockFileItem, adding each exsisting file as a property as a LockFileItem.
+So the targets section in assets file is as following:
+```json
+{
+  "version": 3,
+  "targets": {
+    ".NETCoreApp,Version=v5.0": {
+      "PackageA/1.0.0": {
+        "type": "package",
+        "compile": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "runtime": {
+           "lib/netstandard2.0/PackageA.dll": {}
+        },
+        "symbol": {
+           "lib/netstandard2.0/PackageA.pdb": {}
+        },
+        "xml": {
+           "lib/netstandard2.0/PackageA.xml": {}
+        },
+    }
+  },
+```
+Pros:
+* It's as powerful as adding existing extension as a property of LockFileItem.
+
+Cons:
+* .pdb and .xml files are not on the same level as compile or runtime items.
+The usage of .pdb and .xml files are SDK driven rather than package author driven.
```

Comment:
What does this mean? It seems like either way if the package author decides to include the files they'll be consumed by the SDK and present in the output the same way, right?

Reply:
Thanks! 
If rephrasing it as the following, is it better? Or worse? :D
From the view of a package author, compile or runtime items are the essential elements. While .pdb and .xml files are not on the same level in the understanding. So the way of having related files as a property of compile and runtime items might be in line with the understanding.

Hi @nkolev92 , may I know if you have any other opinions about rephrasing this cons? Thanks!

Reply:
I think my confusion comes from this: if these files are present in the NuGet package, the package author had to choose to put them there (right?). So they clearly care about it and expect package consumers to use the files somehow.